specutils/core/generic.py

```diff
@@ -0,0 +1,124 @@
+from specutils.spectrum1d import Spectrum1D
+
+import numpy as np
+import logging
+
+from astropy.nddata import NDArithmeticMixin, NDSlicingMixin, NDIOMixin
+import astropy.units as u
+from astropy.wcs import WCS
+
+
+class GenericSpectrum1D(NDIOMixin, NDSlicingMixin, NDArithmeticMixin,
+                        Spectrum1D):
+    """
+    A generalized spectrum data object that implements several `NDData`
+    features such as IO, arithmetic, and slicing.
```

Comment:
I think it would be useful to add just a bit more here about the differences.  In particular it should be noted here that the wcs is *different* from the `Spectrum1D` wcs.